### PR TITLE
fix: add env variable KAFKA_CFG_MESSAGE_MAX_BYTES

### PR DESCRIPTION
**Pull request by jk2K**
when adjust message max size, need to change message.max.bytes and max.request.size

### CODE REVIEW

**Comment by carrodher** [comment]
Thank you for initiating this pull request. We appreciate your effort. Just a friendly reminder that it's important to sign your commits. Adding your signature certifies that you either authored the patch or have the necessary rights to contribute the changes. You can find detailed information on how to do this in the [“Sign your work” section of our contributing guidelines](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work).

Feel free to reach out if you have any questions or need assistance with the signing process.

**Comment by jk2K** [comment]
> Thank you for initiating this pull request. We appreciate your effort. Just a friendly reminder that it's important to sign your commits. Adding your signature certifies that you either authored the patch or have the necessary rights to contribute the changes. You can find detailed information on how to do this in the [“Sign your work” section of our contributing guidelines](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work).
> 
> Feel free to reach out if you have any questions or need assistance with the signing process.

signed

**Comment by Mauraza** [comment]
Hi @jk2K,

Thanks for the contribution!! I'm going to run some tests, and then I will update this thread.

File: bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/kafka-env.sh [hunk 1]
```diff
@@ -95,6 +95,9 @@ export KAFKA_OPTS="${KAFKA_OPTS:-}"
 export KAFKA_CFG_SASL_ENABLED_MECHANISMS="${KAFKA_CFG_SASL_ENABLED_MECHANISMS:-PLAIN,SCRAM-SHA-256,SCRAM-SHA-512}"
 export KAFKA_KRAFT_CLUSTER_ID="${KAFKA_KRAFT_CLUSTER_ID:-}"
 export KAFKA_SKIP_KRAFT_STORAGE_INIT="${KAFKA_SKIP_KRAFT_STORAGE_INIT:-false}"
+export KAFKA_CFG_MAX_REQUEST_SIZE="${KAFKA_CFG_MAX_REQUEST_SIZE:-1048576}"
+export KAFKA_CFG_MESSAGE_MAX_BYTES="${KAFKA_CFG_MESSAGE_MAX_BYTES:-1048576}"
+export KAFKA_CFG_MAX_PARTITION_FETCH_BYTES="${KAFKA_CFG_MAX_PARTITION_FETCH_BYTES:-1048576}"
```
Comment (Mauraza) [inline bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/kafka-env.sh:100]: Could you tell me why 1048576 is the default value?

File: bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh [hunk 2]
```diff
@@ -937,6 +937,7 @@ kafka_initialize() {
         kafka_configure_from_environment_variables
         # Configure Kafka producer/consumer to set up message sizes
         ! is_empty_value "${KAFKA_CFG_MAX_REQUEST_SIZE:-}" && kafka_common_conf_set "$KAFKA_CONF_DIR/producer.properties" max.request.size "$KAFKA_CFG_MAX_REQUEST_SIZE"
+        ! is_empty_value "${KAFKA_CFG_MESSAGE_MAX_BYTES:-}" && kafka_server_conf_set message.max.bytes "$KAFKA_CFG_MESSAGE_MAX_BYTES"
         ! is_empty_value "${KAFKA_CFG_MAX_PARTITION_FETCH_BYTES:-}" && kafka_common_conf_set "$KAFKA_CONF_DIR/consumer.properties" max.partition.fetch.bytes "$KAFKA_CFG_MAX_PARTITION_FETCH_BYTES"
```
Comment (Mauraza) [inline bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:941]: If these variables have a default value, they shouldn't be empty.

**Review by Mauraza — COMMENTED** [review]
Hi @jk2K,

Could you be so kind as to review the comments?

File: bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/kafka-env.sh (hunk 1 above)
Comment (jk2K) [reply bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/kafka-env.sh:100]: https://kafka.apache.org/documentation/#producerconfigs_max.request.size


Default: | 1048576

File: bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh (hunk 2 above)
Comment (jk2K) [reply bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:941]: keep the code style consistent

Comment (Mauraza) [reply bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:941]: I think we can remove these lines. 
```suggestion
```

File: bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh [hunk 3]
```diff
@@ -937,6 +937,7 @@ kafka_initialize() {
         kafka_configure_from_environment_variables
         # Configure Kafka producer/consumer to set up message sizes
         ! is_empty_value "${KAFKA_CFG_MAX_REQUEST_SIZE:-}" && kafka_common_conf_set "$KAFKA_CONF_DIR/producer.properties" max.request.size "$KAFKA_CFG_MAX_REQUEST_SIZE"
+        ! is_empty_value "${KAFKA_CFG_MESSAGE_MAX_BYTES:-}" && kafka_server_conf_set message.max.bytes "$KAFKA_CFG_MESSAGE_MAX_BYTES"
         ! is_empty_value "${KAFKA_CFG_MAX_PARTITION_FETCH_BYTES:-}" && kafka_common_conf_set "$KAFKA_CONF_DIR/consumer.properties" max.partition.fetch.bytes "$KAFKA_CFG_MAX_PARTITION_FETCH_BYTES"
```
Comment (Mauraza) [inline bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:941]:
```suggestion
```

File: bitnami/kafka/3.4/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh [hunk 4]
```diff
@@ -937,6 +937,7 @@ kafka_initialize() {
         kafka_configure_from_environment_variables
         # Configure Kafka producer/consumer to set up message sizes
         ! is_empty_value "${KAFKA_CFG_MAX_REQUEST_SIZE:-}" && kafka_common_conf_set "$KAFKA_CONF_DIR/producer.properties" max.request.size "$KAFKA_CFG_MAX_REQUEST_SIZE"
+        ! is_empty_value "${KAFKA_CFG_MESSAGE_MAX_BYTES:-}" && kafka_server_conf_set message.max.bytes "$KAFKA_CFG_MESSAGE_MAX_BYTES"
         ! is_empty_value "${KAFKA_CFG_MAX_PARTITION_FETCH_BYTES:-}" && kafka_common_conf_set "$KAFKA_CONF_DIR/consumer.properties" max.partition.fetch.bytes "$KAFKA_CFG_MAX_PARTITION_FETCH_BYTES"
```
Comment (Mauraza) [inline bitnami/kafka/3.4/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:941]:
```suggestion
```

File: bitnami/kafka/3.5/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh [hunk 5]
```diff
@@ -937,6 +937,7 @@ kafka_initialize() {
         kafka_configure_from_environment_variables
         # Configure Kafka producer/consumer to set up message sizes
         ! is_empty_value "${KAFKA_CFG_MAX_REQUEST_SIZE:-}" && kafka_common_conf_set "$KAFKA_CONF_DIR/producer.properties" max.request.size "$KAFKA_CFG_MAX_REQUEST_SIZE"
+        ! is_empty_value "${KAFKA_CFG_MESSAGE_MAX_BYTES:-}" && kafka_server_conf_set message.max.bytes "$KAFKA_CFG_MESSAGE_MAX_BYTES"
         ! is_empty_value "${KAFKA_CFG_MAX_PARTITION_FETCH_BYTES:-}" && kafka_common_conf_set "$KAFKA_CONF_DIR/consumer.properties" max.partition.fetch.bytes "$KAFKA_CFG_MAX_PARTITION_FETCH_BYTES"
```
Comment (Mauraza) [inline bitnami/kafka/3.5/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:941]:
```suggestion
```

File: bitnami/kafka/3.6/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh [hunk 6]
```diff
@@ -937,6 +937,7 @@ kafka_initialize() {
         kafka_configure_from_environment_variables
         # Configure Kafka producer/consumer to set up message sizes
         ! is_empty_value "${KAFKA_CFG_MAX_REQUEST_SIZE:-}" && kafka_common_conf_set "$KAFKA_CONF_DIR/producer.properties" max.request.size "$KAFKA_CFG_MAX_REQUEST_SIZE"
+        ! is_empty_value "${KAFKA_CFG_MESSAGE_MAX_BYTES:-}" && kafka_server_conf_set message.max.bytes "$KAFKA_CFG_MESSAGE_MAX_BYTES"
         ! is_empty_value "${KAFKA_CFG_MAX_PARTITION_FETCH_BYTES:-}" && kafka_common_conf_set "$KAFKA_CONF_DIR/consumer.properties" max.partition.fetch.bytes "$KAFKA_CFG_MAX_PARTITION_FETCH_BYTES"
```
Comment (Mauraza) [inline bitnami/kafka/3.6/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:941]:
```suggestion
```

File: bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh (hunk 2 above)
Comment (jk2K) [reply bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:941]: remove all 3 lines?

Comment (Mauraza) [reply bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:941]: Sorry, I messed up. I just updated the comment.

File: bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/kafka-env.sh [hunk 7]
```diff
@@ -95,6 +95,9 @@ export KAFKA_OPTS="${KAFKA_OPTS:-}"
 export KAFKA_CFG_SASL_ENABLED_MECHANISMS="${KAFKA_CFG_SASL_ENABLED_MECHANISMS:-PLAIN,SCRAM-SHA-256,SCRAM-SHA-512}"
 export KAFKA_KRAFT_CLUSTER_ID="${KAFKA_KRAFT_CLUSTER_ID:-}"
 export KAFKA_SKIP_KRAFT_STORAGE_INIT="${KAFKA_SKIP_KRAFT_STORAGE_INIT:-false}"
+export KAFKA_CFG_MAX_REQUEST_SIZE="${KAFKA_CFG_MAX_REQUEST_SIZE:-1048576}"
+export KAFKA_CFG_MESSAGE_MAX_BYTES="${KAFKA_CFG_MESSAGE_MAX_BYTES:-1048576}"
+export KAFKA_CFG_MAX_PARTITION_FETCH_BYTES="${KAFKA_CFG_MAX_PARTITION_FETCH_BYTES:-1048576}"
```
Comment (Mauraza) [inline bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/kafka-env.sh:100]:
```suggestion
export KAFKA_CFG_MAX_REQUEST_SIZE="${KAFKA_CFG_MAX_REQUEST_SIZE:-}"
export KAFKA_CFG_MESSAGE_MAX_BYTES="${KAFKA_CFG_MESSAGE_MAX_BYTES:-}"
export KAFKA_CFG_MAX_PARTITION_FETCH_BYTES="${KAFKA_CFG_MAX_PARTITION_FETCH_BYTES:-}"
```

File: bitnami/kafka/3.5/debian-11/rootfs/opt/bitnami/scripts/kafka-env.sh [hunk 8]
```diff
@@ -95,6 +95,9 @@ export KAFKA_OPTS="${KAFKA_OPTS:-}"
 export KAFKA_CFG_SASL_ENABLED_MECHANISMS="${KAFKA_CFG_SASL_ENABLED_MECHANISMS:-PLAIN,SCRAM-SHA-256,SCRAM-SHA-512}"
 export KAFKA_KRAFT_CLUSTER_ID="${KAFKA_KRAFT_CLUSTER_ID:-}"
 export KAFKA_SKIP_KRAFT_STORAGE_INIT="${KAFKA_SKIP_KRAFT_STORAGE_INIT:-false}"
+export KAFKA_CFG_MAX_REQUEST_SIZE="${KAFKA_CFG_MAX_REQUEST_SIZE:-1048576}"
+export KAFKA_CFG_MESSAGE_MAX_BYTES="${KAFKA_CFG_MESSAGE_MAX_BYTES:-1048576}"
+export KAFKA_CFG_MAX_PARTITION_FETCH_BYTES="${KAFKA_CFG_MAX_PARTITION_FETCH_BYTES:-1048576}"
```
Comment (Mauraza) [inline bitnami/kafka/3.5/debian-11/rootfs/opt/bitnami/scripts/kafka-env.sh:100]:
```suggestion
export KAFKA_CFG_MAX_REQUEST_SIZE="${KAFKA_CFG_MAX_REQUEST_SIZE:-}"
export KAFKA_CFG_MESSAGE_MAX_BYTES="${KAFKA_CFG_MESSAGE_MAX_BYTES:-}"
export KAFKA_CFG_MAX_PARTITION_FETCH_BYTES="${KAFKA_CFG_MAX_PARTITION_FETCH_BYTES:-}"
```

File: bitnami/kafka/3.6/debian-11/rootfs/opt/bitnami/scripts/kafka-env.sh [hunk 9]
```diff
@@ -95,6 +95,9 @@ export KAFKA_OPTS="${KAFKA_OPTS:-}"
 export KAFKA_CFG_SASL_ENABLED_MECHANISMS="${KAFKA_CFG_SASL_ENABLED_MECHANISMS:-PLAIN,SCRAM-SHA-256,SCRAM-SHA-512}"
 export KAFKA_KRAFT_CLUSTER_ID="${KAFKA_KRAFT_CLUSTER_ID:-}"
 export KAFKA_SKIP_KRAFT_STORAGE_INIT="${KAFKA_SKIP_KRAFT_STORAGE_INIT:-false}"
+export KAFKA_CFG_MAX_REQUEST_SIZE="${KAFKA_CFG_MAX_REQUEST_SIZE:-1048576}"
+export KAFKA_CFG_MESSAGE_MAX_BYTES="${KAFKA_CFG_MESSAGE_MAX_BYTES:-1048576}"
+export KAFKA_CFG_MAX_PARTITION_FETCH_BYTES="${KAFKA_CFG_MAX_PARTITION_FETCH_BYTES:-1048576}"
```
Comment (Mauraza) [inline bitnami/kafka/3.6/debian-11/rootfs/opt/bitnami/scripts/kafka-env.sh:100]:
```suggestion
export KAFKA_CFG_MAX_REQUEST_SIZE="${KAFKA_CFG_MAX_REQUEST_SIZE:-}"
export KAFKA_CFG_MESSAGE_MAX_BYTES="${KAFKA_CFG_MESSAGE_MAX_BYTES:-}"
export KAFKA_CFG_MAX_PARTITION_FETCH_BYTES="${KAFKA_CFG_MAX_PARTITION_FETCH_BYTES:-}"
```

**Review by Mauraza — APPROVED** [review]
Hi @jk2K,

Thanks for the contribution!! this will be available in the next revision of Kafka.

**Comment by fmulero** [comment]
Hi @jk2K  We reverted these changes due to the issue reported on  https://github.com/bitnami/charts/issues/22552

Could you share why you needed these changes? You can set the env var `KAFKA_CFG_MESSAGE_MAX_BYTES` without them and the kafka property `message.max.properties` will be set. This function will do it for us: https://github.com/bitnami/containers/blob/f8f75159d5e259900b949a84622dd8f8cdd5b3a0/bitnami/kafka/3.4/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh#L791-L821

**Comment by jk2K** [comment]
my message is big, exceeded the default 1MB of [message.max.bytes](https://kafka.apache.org/documentation/#brokerconfigs_message.max.bytes)

**Comment by fmulero** [comment]
You could do it just setting the env var, for instance `KAFKA_CFG_MESSAGE_MAX_BYTES =5242940`. You didn't need these changes. Please have a look to this example with the tag `[3.5.2-debian-11-r1](https://hub.docker.com/layers/bitnami/kafka/3.5.2-debian-11-r1/images/sha256-1d447c670c074ec8264f863e0590f81b057cd717f38b7e05608a2e75e1ecec7e?context=explore)` (this tag doesn't have your changes):

```yaml
version: "2"

services:
  kafka:
    image: bitnami/kafka:3.5.2-debian-11-r1
    ports:
      - "9092:9092"
    volumes:
      - "kafka_data:/bitnami"
    environment:
      # KRaft settings
      - KAFKA_CFG_NODE_ID=0
      - KAFKA_CFG_PROCESS_ROLES=controller,broker
      - KAFKA_CFG_CONTROLLER_QUORUM_VOTERS=0@kafka:9093
      - KAFKA_CFG_MESSAGE_MAX_BYTES=5242940
      # Listeners
      - KAFKA_CFG_LISTENERS=PLAINTEXT://:9092,CONTROLLER://:9093
      - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://:9092
      - KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT
      - KAFKA_CFG_CONTROLLER_LISTENER_NAMES=CONTROLLER
      - KAFKA_CFG_INTER_BROKER_LISTENER_NAME=PLAINTEXT
volumes:
  kafka_data:
    driver: local
```

If we start that docker-compose and check the `/opt/bitnami/kafka/config/server.properties` we can see this:
```console
$ docker-compose up -d
$ docker-compose exec kafka tail -n 25 /opt/bitnami/kafka/config/server.properties

############################# Group Coordinator Settings #############################

# The following configuration specifies the time, in milliseconds, that the GroupCoordinator will delay the initial consumer rebalance.
# The rebalance will be further delayed by the value of group.initial.rebalance.delay.ms as new members join the group, up to a maximum of max.poll.interval.ms.
# The default value for this is 3 seconds.
# We override this to 0 here as it makes for a better out-of-the-box experience for development and testing.
# However, in production environments the default value of 3 seconds is more suitable as this will help to avoid unnecessary, and potentially expensive, rebalances during application startup.
#group.initial.rebalance.delay.ms=0

controller.listener.names=CONTROLLER
controller.quorum.voters=0@kafka:9093
inter.broker.listener.name=PLAINTEXT

message.max.bytes=5242940
node.id=0
process.roles=controller,broker
sasl.enabled.mechanisms=PLAIN,SCRAM-SHA-256,SCRAM-SHA-512
```

**Comment by jk2K** [comment]
ok, I will try it, thanks

**Pull request merged**